el-patch.el

```diff
@@ -335,12 +335,13 @@ be defined permanently."
 (defun el-patch--classify-definition-type (type)
   "Classifies a definition TYPE as a `function' or `variable'.
 TYPE is a symbol `defun', `defmacro', etc."
-  (pcase type
-    ((or 'defun 'defmacro 'defsubst 'define-minor-mode)
-     'function)
-    ((or 'defvar 'defconst 'defcustom)
-     'variable)
-    (_ (error "Unexpected definition type %S" type))))
+  (let ((classify (assoc type el-patch-deftype-alist)))
+    (if classify
+        (pcase (nth 2 classify)
```

Comment:
The values in `el-patch-deftype-alist` are plists. You can't assume the order of the keywords will always be the same; use `plist-get` instead.